File: Emulator/src/main/content/global/skill/summoning/familiar/dialogue/DismissDialogue.java
package content.global.skill.summoning.familiar.dialogue;

import core.game.dialogue.Dialogue;
import core.game.dialogue.DialogueInterpreter;
import core.game.node.entity.player.Player;
import core.plugin.Initializable;

import static core.api.ContentAPIKt.sendDialogueOptions;

/**
 * The type Dismiss dialogue.
 */
@Initializable
public class DismissDialogue extends Dialogue {

    @Override
    public Dialogue newInstance(Player player) {
        return new DismissDialogue(player);
    }

    /**
     * Instantiates a new Dismiss dialogue.
     */
    public DismissDialogue() {}

    /**
     * Instantiates a new Dismiss dialogue.
     *
     * @param player the player
     */
    public DismissDialogue(Player player) {
        super(player);
    }

    @Override
    public boolean open(Object... args) {
        if (player.getFamiliarManager().getFamiliar() instanceof content.global.skill.summoning.pet.Pet) {
            sendDialogueOptions(player, "Free pet?", "Yes", "No");
        } else {
            sendDialogueOptions(player, "Dismiss Familiar?", "Yes", "No");
        }
        return true;
    }

    @Override
    public boolean handle(int interfaceId, int buttonId) {
        switch (stage) {
            case 0:
                if (buttonId == 1) {
                    if (player.getFamiliarManager().getFamiliar() instanceof content.global.skill.summoning.pet.Pet) {
                        sendDialogue("Run along; I'm setting you free.");
                        content.global.skill.summoning.pet.Pet pet = (content.global.skill.summoning.pet.Pet) player.getFamiliarManager().getFamiliar();
                        player.getFamiliarManager().removeDetails(pet.getItemIdHash());
                    } else {
                        end();
                    }
                    player.getFamiliarManager().dismiss();
                    stage = 1;
                } else if (buttonId == 2) {
                    end();
                }
                break;
            case 1:
                end();
                break;
        }
        return true;
    }

    @Override
    public int[] getIds() {
        return new int[]{DialogueInterpreter.getDialogueKey("dismiss_dial")};
    }
}

File: Emulator/src/main/content/global/skill/summoning/familiar/dialogue/DismissDialogue.kt
package content.global.skill.summoning.familiar.dialogue

import content.global.skill.summoning.pet.Pet
import core.api.sendDialogueOptions
import core.game.dialogue.Dialogue
import core.game.dialogue.DialogueInterpreter
import core.game.node.entity.player.Player
import core.plugin.Initializable

/**
 * The type Dismiss dialogue.
 */
@Initializable
class DismissDialogue : Dialogue {
    override fun newInstance(player: Player): Dialogue {
        return DismissDialogue(player)
    }

    /**
     * Instantiates a new Dismiss dialogue.
     */
    constructor()

    /**
     * Instantiates a new Dismiss dialogue.
     *
     * @param player the player
     */
    constructor(player: Player?) : super(player)

    override fun open(vararg args: Any): Boolean {
        if (player.familiarManager.familiar is Pet) {
            sendDialogueOptions(player, "Free pet?", "Yes", "No")
        } else {
            sendDialogueOptions(player, "Dismiss Familiar?", "Yes", "No")
        }
        return true
    }

    override fun handle(interfaceId: Int, buttonId: Int): Boolean {
        when (stage) {
            0 -> if (buttonId == 1) {
                if (player.familiarManager.familiar is Pet) {
                    sendDialogue("Run along; I'm setting you free.")
                    val pet = player.familiarManager.familiar as Pet
                    player.familiarManager.removeDetails(pet.itemIdHash)
                } else {
                    end()
                }
                player.familiarManager.dismiss()
                stage = 1
            } else if (buttonId == 2) {
                end()
            }

            1 -> end()
        }
        return true
    }

    override fun getIds(): IntArray {
        return intArrayOf(DialogueInterpreter.getDialogueKey("dismiss_dial"))
    }
}
